### DISCUSSION TITLE
Can't declare variables in for loops

### DISCUSSION THREAD

**Discussion by ltecheroffical** [cc65/cc65#2481] [General]
When using cc65, I found myself unable to declare variables in for loops for example

This code:

```c
for (char i = 0; i < 32; i++) {
    // ...
}
```
Would not compile while in other compilers this almost always compiles

As well as the fact that this won't compile either:

```c
void function() {
    // Some code here
    
    char i = 0;
    for (i = 0; i < 32; i++) {
        // ...
    }
}
```

Platform: MacOS
CC65: cc65 V2.19 - Git feb502682

**Comment by spiro-trikaliotis** [comment]
cc65 is mainly a C89/C90 compiler, with small additions here and there from newer standards.

The ability to define variables in the for statement as well as defining variables anywhere but at the beginning of a block was introduced in C99, though.

Thus, what you are asking for is an enhancement, but clearly not a bug.

**Comment by ltecheroffical** [comment]
> cc65 is mainly a C89/C90 compiler, with small additions here and there from newer standards.
> 
> The ability to define variables in the for statement as well as defining variables anywhere but at the beginning of a block was introduced in C99, though.
> 
> Thus, what you are asking for is an enhancement, but clearly not a bug.

Ah, I was working in C99 codebases mostly and there I was fully able to declare normal syntax

**Comment by polluks** [comment]
However we could drop this limitation with `--standard c99`.

**Comment by ltecheroffical** [comment]
> However we could drop this limitation with `--standard c99`.

Maybe the limitation should also be dropped with `--standard cc65` as well, c99 doesn't have a lot of things that would really clean up the code on older systems using every bit possible, mainly the `0b00000000` syntax

**Comment by spiro-trikaliotis** [comment]
> However we could drop this limitation with `--standard c99`.

Of course, feel free to work on it. ;)

**Comment by ltecheroffical** [comment]
Also maybe implement it to the cc65 standard
